lib/common/xpath.c

```diff
@@ -56,6 +56,94 @@ freeXpathObject(xmlXPathObjectPtr xpathObj)
     xmlXPathFreeObject(xpathObj);
 }
 
+
+/*!
+ * \internal
+ * \brief Get an element node from the result of evaluating an XPath expression
+ *
+ * Evaluating an XPath expression stores the list of matching nodes in an
+ * \c xmlXPathObject. This function gets the node at a particular index within
+ * that list.
+ *
+ * Each matching node may be of an arbitrary type. This function is guaranteed
+ * to return an element node (or \c NULL).
```

Comment:
I might drop this logic in a later commit. Note to self, this would mean `getXpathResult()` needs to be updated to use this logic explicitly.

It's counterintuitive, and every internal XPath expression that I've spot-checked will only match against elements anyway. The one exception I've found is `cib_process_xpath()`. It looks like if the user requests deletion of an attribute via an XPath expression, the attribute's parent will get deleted instead. And as much as we care about backward compatibility, that seems like pretty clearly a bug.

Reply:
Done -- updated commits. Commits prior to "Refactor: libstonithd: Query stonith devices more clearly" are the same I think.